test/functional/Java8andUp/src/org/openj9/test/classRelationshipVerifier/TestClassRelationshipVerifier.java

```diff
@@ -0,0 +1,252 @@
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package org.openj9.test.classRelationshipVerifier;
+
+import java.lang.reflect.*;
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import org.objectweb.asm.*;
+
+/**
+ * Definitions:
+ * 	- Caller: the calling class, represented by A
+ * 	- Source: the child class, represented by B
+ * 	- Target: the parent class/interface, represented by C
+ */
+
+@Test(groups = { "level.sanity" })
+public class TestClassRelationshipVerifier {
+	static {
+		try {
+			System.loadLibrary("bcvrelationships");
+		} catch (UnsatisfiedLinkError e) {
+			Assert.fail(e.getMessage() + "\nlibrary path = " + System.getProperty("java.library.path"));
+		}
+	}
+
+	native static boolean isRelationshipRecorded(String sourceClassName, String targetClassName, ClassLoader classLoader);
+
+	/**
+	 * Target class is not loaded, but source class is loaded.
+	 * Check that a relationship is recorded for the source and target classes.
+	 *
+	 * @throws Throwable
+	 */
+	@Test
+	static public void testOnlySourceLoaded() throws Throwable {
+		CustomClassLoader classLoader = new CustomClassLoader();
+
+		byte[] bytesClassA = ClassGenerator.classA_dump();
+		byte[] bytesClassB = ClassGenerator.classB_dump();
+
+		Class<?> clazzA = classLoader.getClass("A", bytesClassA);
+		Class<?> clazzB = classLoader.getClass("B", bytesClassB);
+
+		try {
+			Object instanceB = clazzB.getDeclaredConstructor().newInstance();
+			Object instanceA = clazzA.getDeclaredConstructor().newInstance();
+
+			if (!isRelationshipRecorded("B", "C", classLoader)) {
+				Assert.fail("relationship was not recorded");
+			}
+		} catch (InvocationTargetException e) {
+			throw e.getCause();
+		}
+	}
+
+	/**
+	 * Source class is not loaded, but target class is loaded.
+	 * Check that a relationship is recorded for the source and target classes.
+	 *
+	 * @throws Throwable
+	 */
+	@Test
+	static public void testOnlyTargetLoaded() throws Throwable {
+		CustomClassLoader classLoader = new CustomClassLoader();
+
+		byte[] bytesClassA = ClassGenerator.classA_dump();
+		byte[] bytesClassC = ClassGenerator.classC_dump();
+
+		Class<?> clazzA = classLoader.getClass("A", bytesClassA);
+		Class<?> clazzC = classLoader.getClass("C", bytesClassC);
+
+		try {
+			Object instanceC = clazzC.getDeclaredConstructor().newInstance();
+			Object instanceA = clazzA.getDeclaredConstructor().newInstance();
+
+			if (!isRelationshipRecorded("B", "C", classLoader)) {
+				Assert.fail("relationship was not recorded");
+			}
+		} catch (InvocationTargetException e) {
+			throw e.getCause();
+		}
+	}
+
+	/**
+	 * Neither source, not target classes are loaded.
```

Comment:
typo: not --> nor

Reply:
Reviewing your own PRs, I see! That's great, but please let the win32 build finish before pushing anything new.